webapp/channels/src/components/header_footer_route/header.tsx

```diff
@@ -21,12 +21,15 @@ export type HeaderProps = {
 const Header = ({alternateLink, backButtonURL, onBackButtonClick}: HeaderProps) => {
     const {EnableCustomBrand, SiteName} = useSelector(getConfig);
 
+    const ariaLabel = EnableCustomBrand === 'true' && SiteName ? SiteName : 'Mattermost';
+
     return (
         <div className='hfroute-header'>
             <div className='header-main'>
                 <Link
                     className='header-logo-link'
                     to='/'
+                    aria-label={ariaLabel}
```

Comment:
This fixes the issue of `Ensures buttons have discernible text `. See https://accessibilityinsights.io/info-examples/web/link-name/